### PR TITLE
[docs] Update the contribution guide

### LINKED ISSUE #4696: [docs] Improve the contribution guide to cover testing and pull request requirements

### LINKED ISSUE #5195: [docs] Add build instructions to TestCafe & Hammerhead contribution guide
Our contribution guide specifies how to run tests, but doesn't contain general building instructions.

https://stackoverflow.com/questions/60869680/how-do-i-get-testcafe-hammerhead-to-build-so-that-i-can-contribute-to-it

### LINKED ISSUE #5441: [docs] Add some points to the contribution guide
I feel that we have missed some things:

Node.js is a prerequisite and needs to be installed on your machine. You need to run `npm install` or `yarn` in the local copy of the repository after it is cloned. You can use `gulp build` to build the project without running tests. After the project is built, you can start TestCafe by executing `node bin/testcafe.js <TestCafe args>` in your terminal.

### PR DESCRIPTION
**Pull request by DIRECTcut**
fixes #5441 
fixes #5195 
fixes #4696 

[hammerhead](https://github.com/DevExpress/testcafe-hammerhead) probably needs similar changes

### CODE REVIEW

**Comment by DIRECTcut** [comment]
@VasilyStrelyaev the `new-docs` branch we mention in the [Documentation for New Features](https://github.com/DevExpress/testcafe/pull/5856/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R116) section of this guideline hasn't received any updates in years. Is this section relevant, or should I modify/remove it completely?

**Comment by VasilyStrelyaev** [comment]
Yes, it's relevant. In fact, it's not quite good that we're not using `new-docs`. I recently published new documentation by accident before the release because I merged it to `master` rather than `new-docs`.

File: CONTRIBUTING.md
Comment (VasilyStrelyaev) [inline CONTRIBUTING.md:59]: That's ambiguous. Do you mean without running *tests that test TestCafe*? (because you do run user's tests from `./tests` in this example).

File: CONTRIBUTING.md
Comment (VasilyStrelyaev) [inline CONTRIBUTING.md:99]: Is this phrase necessary?

File: CONTRIBUTING.md
Comment (VasilyStrelyaev) [reply CONTRIBUTING.md:59]: Or... Are *tests that test TestCafe* in `./tests`? Why `without running tests` then?

You see, I'm puzzled (honestly) 😄

Comment (DIRECTcut) [reply CONTRIBUTING.md:59]: You're right, that's ambiguous. What about:

```text
 You can build TestCafe and launch it to test changes manually:
```

File: CONTRIBUTING.md
Comment (DIRECTcut) [reply CONTRIBUTING.md]: Removed it.

**Review by VasilyStrelyaev — APPROVED** [review]

File: CONTRIBUTING.md
Comment (AlexSkorkin) [inline CONTRIBUTING.md:92]:
```suggestion
* All existing and new tests must be passing
```

Probably worth clarifying what tests must pass.

File: CONTRIBUTING.md
Comment (AlexSkorkin) [inline CONTRIBUTING.md:95]:
```suggestion
Please keep in mind that the team may **suspend or reject** pull requests that fail to meet these requirements.
```
Not sure if 'might' is suitable here.

**Review by AlexSkorkin — COMMENTED** [review]
Looks good, you might want to check out my comments. No need to wait my approval.

File: CONTRIBUTING.md
Comment (DIRECTcut) [reply CONTRIBUTING.md:91]: Yes, this is an actual keyword. To link an `issue` to a `pull request`, users write the following text in the body of said `pull request`:

`closes #1234`

 [Reference](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: CONTRIBUTING.md
Comment (DIRECTcut) [reply CONTRIBUTING.md:92]: Tests can either pass or fail. Before I submit a PR, all tests need to pass.

File: CONTRIBUTING.md
Comment (arminal) [inline CONTRIBUTING.md:28]:
```suggestion
When you create a new issue, the template text is automatically added to its body. Complete all the sections in this template to help us understand the problem you are describing. Missing information could delay processing time.
```

File: CONTRIBUTING.md
Comment (arminal) [inline CONTRIBUTING.md:34]:
```suggestion
1. Search the [list of issues](https://github.com/DevExpress/testcafe/issues) to see if there is an issue for the bug or feature you plan to work on. If none exists, create a new one.
```

File: CONTRIBUTING.md
Comment (arminal) [reply CONTRIBUTING.md:91]: ok

File: CONTRIBUTING.md
Comment (arminal) [reply CONTRIBUTING.md:92]: Before you submit a PR, launch tests, all of which must pass before you continue.

File: CONTRIBUTING.md
Comment (DIRECTcut) [inline CONTRIBUTING.md:92]:
```suggestion
* Run tests, all of which must pass before you continue.
```

File: CONTRIBUTING.md
Comment (arminal) [inline CONTRIBUTING.md:125]:
```suggestion
The `/lib` directory stores build artifacts. Build tasks remove this folder before they run. To remove a folder manually, run:
```

Comment (DIRECTcut) [reply CONTRIBUTING.md:125]: Is this OK:

> The `/lib` directory stores build artifacts. Build tasks remove this folder before they run. To remove **this** folder manually, run:

Comment (arminal) [reply CONTRIBUTING.md:125]: yes

Comment (DIRECTcut) [inline CONTRIBUTING.md:125]:
```suggestion
The `/lib` directory stores build artifacts. Build tasks remove this folder before they run. To remove this directory manually, run:
```

**Pull request merged**
